### ISSUE TITLE
[Bug]: After closing all groups, opening any buffer and try to pin, it throws an error

### ISSUE THREAD

**Issue by otavioschwanck** [akinsho/bufferline.nvim#692]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

<img width="1288" alt="image" src="https://user-images.githubusercontent.com/9551316/220371555-5d8aa6fa-a4fd-4d96-8ae4-1da787521f58.png">

### What did you expect to happen?

Just pin

### Config

```lua
return {
  {
    "akinsho/bufferline.nvim", branch = 'dev',
    -- dir =  "~/Projetos/bufferline.nvim/",
    lazy = false,
    keys = {
      { "<leader>bp", "<Cmd>BufferLineTogglePin<CR>",                                                desc = "Toggle pin" },
      { "<leader>bk", "<Cmd>BufferLineGroupClose ungrouped<CR>",                                     desc = "Delete non-pinned buffers" },
      { "<leader>bK", "<Cmd>BufferLineGroupClose ungrouped<CR><Cmd>BufferLineGroupClose pinned<CR>", desc = "Delete all buffers" },
      { "<leader>bn", ":e ~/.nvim-scratch<CR>",                                                      desc = "Open Scratch Buffer" },
      { "<C-s>",      "<cmd>BufferLineTogglePin<cr>",                                                desc = "Prev Buffer" },
      { "[b",         "<cmd>BufferLineCyclePrev<cr>",                                                desc = "Prev Buffer" },
      { "]b",         "<cmd>BufferLineCycleNext<cr>",                                                desc = "Next Buffer" },
      { "H",          "<cmd>BufferLineCyclePrev<cr>",                                                desc = "Prev Buffer" },
      { "L",          "<cmd>BufferLineCycleNext<cr>",                                                desc = "Next Buffer" },
      { "[B",         "<cmd>BufferLineMovePrev<cr>",                                                 desc = "Switch Prev Buffer" },
      { "]B",         "<cmd>BufferLineMoveNext<cr>",                                                 desc = "Switch Next Buffer" },
      { "<C-h>",      "<cmd>BufferLineMovePrev<cr>",                                                 desc = "Switch Prev Buffer" },
      { "<C-l>",      "<cmd>BufferLineMoveNext<cr>",                                                 desc = "Switch Next Buffer" },
    },
    opts = {
      options = {
        diagnostics = "nvim_lsp",
        always_show_bufferline = true,
        offsets = {
          {
            filetype = "neo-tree",
            text = "Neo-tree",
            highlight = "Directory",
            text_align = "left",
          },
        },
      },
    },
  }
}
```

### commit

6dbb3d44a6149b1919f7b13e117416fb18f0814b

**Comment by otavioschwanck** [comment]
The commit that makes this error happen:

6dbb3d44a6149b1919f7b13e117416fb18f0814b

**Comment by akinsho** [comment]
@otavioschwanck are you trying to use the `ungrouped` group to handle closing all of your buffers? because you shouldn't be closing *every group* only custom group that you made? or at least I didn't design groups as a way for a user to close all buffers that's not a correct usage.

**Comment by otavioschwanck** [comment]
> @otavioschwanck are you trying to use the `ungrouped` group to handle closing all of your buffers? because you shouldn't be closing _every group_ only custom group that you made? or at least I didn't design groups as a way for a user to close all buffers that's not a correct usage.

Im closing the pinned + ungrouped.  After that, i open a buffer and cannot pin it (throws that error)

**Comment by akinsho** [comment]
I'll have a look when I have time, as always PRs welcome

**Comment by akinsho** [comment]
fixed by #693
